src/lib.rs

```diff
@@ -60,12 +60,27 @@ pub enum Subsample {
     ///
     /// Note that "within" excludes points directly on the zone's boundary.
     RandomPoints,
-    /// Sample uniformly at random from these points within the zone's shape.
+    /// Sample from points within the zone's shape, where each point has a relative weight.
     ///
     /// Note that "within" excludes points directly on the zone's boundary. If a point lies in more
     /// than one zone, it'll be assigned to any of those zones arbitrarily. (This means the input
     /// zones overlap.)
-    UnweightedPoints(Vec<Point<f64>>),
+    WeightedPoints(Vec<WeightedPoint>),
+}
+
+/// A point with an associated relative weight. Higher weights are more likely to be sampled.
+#[derive(Clone)]
+pub struct WeightedPoint {
+    pub point: Point<f64>,
+    pub weight: f64,
+}
+
+impl RTreeObject for WeightedPoint {
```

Comment:
I just followed https://docs.rs/rstar/latest/rstar/trait.RTreeObject.html#example-implementation so we can plug in our new type into the R-tree

Reply:
:+1: so nice.